oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/ColumnTypeEsMapping.java

```diff
@@ -28,13 +28,15 @@
 
 public class ColumnTypeEsMapping {
 
-    public String transform(Class<?> type, Type genericType, final ElasticSearchExtension elasticSearchExtension) {
+    public String transform(Class<?> type, Type genericType, int length, boolean storageOnly, final ElasticSearchExtension elasticSearchExtension) {
         if (Integer.class.equals(type) || int.class.equals(type) || Layer.class.equals(type)) {
             return "integer";
         } else if (Long.class.equals(type) || long.class.equals(type)) {
             return "long";
         } else if (Double.class.equals(type) || double.class.equals(type)) {
             return "double";
+        } else if (String.class.equals(type) && storageOnly && length > 32766) {   // 32766 is the max length of a keywords field in ES
+            return "text";
```

Comment:
@kezhenxu94 @wankai123 This could be a hidden bug, in which length doesn't apply eventually, and fails back to keyword type. The good part is, that this would not break OAP booting.